### PR TITLE
Add ESLint rule: sort-keys

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Adds [ESLint’s sort-keys rule](https://eslint.org/docs/rules/sort-keys).
Objects with more than 5 keys are expected to be sorted.

## Why?

More consistency for ever-growing objects.

## Known issues

There are a few examples, such as ab-tests, where the ordering is consistent, but not alphabetical. The proposed approach is to ignore this rule for those files.

### CODE REVIEW

File: dotcom-rendering/src/web/lib/acquisitions.test.ts [hunk 1]
```diff
@@ -3,20 +3,20 @@ import { addTrackingCodesToUrl } from '@root/src/web/lib/acquisitions';
...
+			campaignCode: 'header_support',
+			componentId: 'header_support',
+			componentType: 'ACQUISITIONS_HEADER',
 			pageViewId: 'abcdefg',
 			referrerUrl: 'https://theguardian.com/uk',
 		});
 
 		expect(result).toEqual(
-			'https://support.theguardian.com/contribute?REFPVID=abcdefg&INTCMP=header_support&acquisitionData=%7B%22source%22%3A%22GUARDIAN_WEB%22%2C%22componentId%22%3A%22header_support%22%2C%22componentType%22%3A%22ACQUISITIONS_HEADER%22%2C%22campaignCode%22%3A%22header_support%22%2C%22abTest%22%3A%7B%22name%22%3A%22testName%22%2C%22variant%22%3A%22variantName%22%7D%2C%22referrerPageviewId%22%3A%22abcdefg%22%2C%22referrerUrl%22%3A%22https%3A%2F%2Ftheguardian.com%2Fuk%22%7D',
+			'https://support.theguardian.com/contribute?REFPVID=abcdefg&INTCMP=header_support&acquisitionData=%7B%22abTest%22%3A%7B%22name%22%3A%22testName%22%2C%22variant%22%3A%22variantName%22%7D%2C%22campaignCode%22%3A%22header_support%22%2C%22componentId%22%3A%22header_support%22%2C%22componentType%22%3A%22ACQUISITIONS_HEADER%22%2C%22referrerPageviewId%22%3A%22abcdefg%22%2C%22referrerUrl%22%3A%22https%3A%2F%2Ftheguardian.com%2Fuk%22%2C%22source%22%3A%22GUARDIAN_WEB%22%7D',
```
Comment (mxdvl) [inline dotcom-rendering/src/web/lib/acquisitions.test.ts:19]: Does this work for the the endpoint to ingest?

**Review by oliverlloyd — COMMENTED** [review]
I think this might cause problems with CAPI data, which is not alphabetically ordered. We use CAPI data for a lot of testing and where we have automated scripts that refresh that this could trip up on this rule (or just cause a lot of shouty lint errors).

We could try to incorporate the lint ignore statements into the refresh scripts but this feel like we're fighting the CAPI tide?

**Comment by AshCorr** [comment]
> We could try to incorporate the lint ignore statements into the refresh scripts but this feel like we're fighting the CAPI tide?

Have the refresh scripts execute `npm prettier:fix`?

**Comment by mxdvl** [comment]
I think we can revisit this if we find it to be a real issue, but I’m going to close it for now.

**Pull request closed without merging**